File: KCD/HMMemberKDockCommand.m
//
//  HMMemberKDockCommand.m
//  KCD
//
//  Created by Hori,Masaki on 2014/02/21.
//  Copyright (c) 2014年 Hori,Masaki. All rights reserved.
//

#import "HMMemberKDockCommand.h"

@implementation HMMemberKDockCommand
+ (void)load
{
	static dispatch_once_t onceToken;
	dispatch_once(&onceToken, ^{
		[HMJSONCommand registerClass:self];
	});
}

+ (BOOL)canExcuteAPI:(NSString *)api
{
	if([api isEqualToString:@"/kcsapi/api_get_member/kdock"]) return YES;
	return NO;
}
- (NSString *)dataKey
{
	if([self.api isEqualToString:@"/kcsapi/api_req_kousyou/getship"]) {
		return @"api_data.api_kdock";
	}
	return [super dataKey];
}
- (void)execute
{
	[self commitJSONToEntityNamed:@"KenzoDock"];
}
@end

File: KCD/HMMemberKDockCommand.swift
//
//  HMMemberKDockCommand.swift
//  KCD
//
//  Created by Hori,Masaki on 2015/01/10.
//  Copyright (c) 2015年 Hori,Masaki. All rights reserved.
//

import Cocoa

class HMMemberKDockCommand: HMJSONCommand {
	required override init() {}
	
	override class func load() {
		HMJSONCommand.registerInstance(self())
	}
	override class func canExcuteAPI(api: String) -> Bool {
		return api == "/kcsapi/api_get_member/kdock"
	}
	
	override var dataKey: String {
		if api == "/kcsapi/api_req_kousyou/getship" {
			return "api_data.api_kdock"
		}
		return super.dataKey
	}
	override func execute() {
		super.commitJSONToEntityNamed("KenzoDock")
	}
}
